### PR TITLE
KOA-4833 Update Line Height and Letter Spacing

### PR DESCRIPTION
**Pull request by frugoman**
Updating Line Height and Letter Spacing to use new values on the font, from foundation tokens.

[x] UNRELEASED.md
[x] Tests
[x] Snapshots

### CODE REVIEW

File: Backpack/Font/Classes/Generated/BPKFont.m [hunk 1]
```diff
@@ -180,6 +226,133 @@ + (UIFont *)fontForStyle:(BPKFontStyle)style fontManager:(BPKFontManager *)fontM
     }
 }
 
++ (NSNumber *)letterSpacingForStyle:(BPKFontStyle)style {
+    switch (style) {
+        case BPKFontStyleTextBase:
```
Comment (marianeum) [inline Backpack/Font/Classes/Generated/BPKFont.m:231]: Are these the legacy letter spacings? I think we need to remove these as they no longer apply (only the hero ones are correct)

File: Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
  * Backpack - Skyscanner's Design System
  *
- * Copyright 2018-2021 Skyscanner Ltd
+ * Copyright 2018-2022 Skyscanner Ltd
```
Comment (marianeum) [inline Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:4]: you're gonna have to do the PR to remove the -2022 bit from the copyright at some point :D

**Review by marianeum — COMMENTED** [review]
might be good to also get a review from an iOS engineer

File: Backpack/Font/Classes/Generated/BPKFont.m (hunk 1 above)
Comment (frugoman) [reply Backpack/Font/Classes/Generated/BPKFont.m:231]: Yes, those are the legacy ones.
Should I remove these and only leave the new ones?

Comment (marianeum) [reply Backpack/Font/Classes/Generated/BPKFont.m:231]: yes let's remove them

File: Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m [hunk 3]
```diff
@@ -167,16 +167,33 @@ - (void)updateTextDisplay {
...
-    NSDictionary<NSAttributedStringKey, id> *newStringAttributes = [BPKFont attributesForFontStyle:self.fontStyle
-                                                                              withCustomAttributes:self.customFontAttributes];
-
-    NSAttributedString *newString = [[NSAttributedString alloc] initWithString:self.text attributes:newStringAttributes];
-    self.contentView.text = newString;
-
+    
+    
+//    NSMutableAttributedString *attText = [[BPKFont attributedStringWithFontStyle:self.fontStyle content:self.text textColor:textColor] mutableCopy];
+//    self.contentView.kern = 0;
```
Comment (RCiesielczuk) [inline Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:173]: Is this needed?

File: Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m [hunk 4]
```diff
@@ -167,16 +167,33 @@ - (void)updateTextDisplay {
...
+    NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+    
+    CGFloat lineHeight = [BPKFont lineHeightForStyle:self.fontStyle];
+    [paragraphStyle setLineSpacing:lineHeight - font.lineHeight];
+    
+    NSMutableDictionary *attributes = [@{
+        NSParagraphStyleAttributeName: paragraphStyle,
+        NSFontAttributeName: font
+    } mutableCopy];
+    if (self.customFontAttributes[NSForegroundColorAttributeName] != nil) {
```
Comment (RCiesielczuk) [inline Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:191]: Do you need to do a null check? I mean, given that this is a dictionary, you'd still be doing a null check when reading, but do you need to do so when writing?

File: Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m (hunk 3 above)
Comment (frugoman) [reply Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:173]: Nope! good catch

**Review by marianeum — APPROVED** [review]

File: Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m (hunk 4 above)
Comment (RCiesielczuk) [reply Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:186]: Are you sure?

Comment (RCiesielczuk) [reply Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m:186]: I take it back :D

**Review by RCiesielczuk — APPROVED** [review]

**Review by marianeum — APPROVED** [review]

**Pull request merged**